examples/bri-3/src/bri/workgroup/worksteps/api/worksteps.controller.spec.ts

```diff
@@ -0,0 +1,193 @@
+      // Arrange
+      const requestDto1 = { name: 'name1', version: 'version1', status: 'status1', workgroupId: 'wgid1', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+      const newWorkstepId1 = await wController.createWorkstep(requestDto1);
+
+      const requestDto2 = { name: 'name2', version: 'version2', status: 'status2', workgroupId: 'wgid2', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+      const newWorkstepId2 = await wController.createWorkstep(requestDto2);
+
+      // Act
+      const worksteps = await wController.getAllWorksteps();
+
+      // Assert
+      expect(worksteps.length).toEqual(2);
+      expect(worksteps[0].id).toEqual(newWorkstepId1);
+      expect(worksteps[0].name).toEqual(requestDto1.name);
+      expect(worksteps[0].version).toEqual(requestDto1.version);
+      expect(worksteps[0].status).toEqual(requestDto1.status);
+      expect(worksteps[0].workgroupId).toEqual(requestDto1.workgroupId);
+      expect(worksteps[0].securityPolicy).toEqual(requestDto1.securityPolicy);
+      expect(worksteps[0].privacyPolicy).toEqual(requestDto1.privacyPolicy);
+      expect(worksteps[1].id).toEqual(newWorkstepId2);
+      expect(worksteps[1].name).toEqual(requestDto2.name);
+      expect(worksteps[1].version).toEqual(requestDto2.version);
+      expect(worksteps[1].status).toEqual(requestDto2.status);
+      expect(worksteps[1].workgroupId).toEqual(requestDto2.workgroupId);
+      expect(worksteps[1].securityPolicy).toEqual(requestDto1.securityPolicy);
+      expect(worksteps[1].privacyPolicy).toEqual(requestDto1.privacyPolicy);
+    });
+  });
+
+  describe('createWorkstep', () => {
+    it('should throw BadRequest if name not provided', () => {
+      // Arrange
+      const requestDto = { version: 'version', status: 'status', workgroupId: 'wgid', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+
+      // Act and assert
+      expect(async () => {
+        await wController.createWorkstep(requestDto);
+      }).rejects.toThrow(new BadRequestException(NAME_EMPTY_ERR_MESSAGE));
+    });
+
+    it('should return new uuid from the created workstep when all params provided', async () => {
+      // Arrange
+      const requestDto = { name: 'name', version: 'version', status: 'status', workgroupId: 'wgid', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+
+      // Act
+      const response = await wController.createWorkstep(requestDto);
+
+      // Assert
+      expect(response.length).toEqual(36);
+    });
+  });
+
+  describe('updateWorkstep', () => {
+    it('should throw NotFound if non existent id passed', () => {
+      // Arrange
+      const nonExistentId = '123';
+      const requestDto = { name: 'name', version: 'version', status: 'status', workgroupId: 'wgid', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as UpdateWorkstepDto;
```

Comment:
const requestDto: UpdateWorkstepDto = {....}
no need for `as UpdateWorkstepDto`

same below

Reply:
moved updateWorkstepsDto to normal typing position